### PR TITLE
Spawn attestation routine at a better place

### PR DESCRIPTION
**Pull request by terencechain**
This PR modified calling `spawnProcessAttestationsRoutine` at a better place. `spawnProcessAttestationsRoutine` is currently called in `startFromSavedState`. There's no logical explanation for why we call it in that function. The downside is that it messes with spec tests with regards to timing since `startFromSavedState` is used in the spec test. To better comply with a single responsibility, we should be calling `spawnProcessAttestationsRoutine` in `Start()` instead

### CODE REVIEW

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -127,6 +127,7 @@ func (s *Service) Start() {
 		if err := s.startFromSavedState(saved); err != nil {
 			log.Fatal(err)
 		}
+		s.spawnProcessAttestationsRoutine(s.cfg.StateNotifier.StateFeed())
```
Comment (kasey) [inline beacon-chain/blockchain/service.go:130]: Is it ok that we'll still call `spawnProcessAttestationsRoutine` from within `startFromPOWChain`?

Comment (kasey) [reply beacon-chain/blockchain/service.go:130]: I'm wondering if this shouldn't be controlled by a `disableProcessAttestationRoutine` flag, or a broader flag that we use for any spec test initialization. This is the first I've heard of `Start` methods being used to isolate logic that shouldn't run in spec tests, in fact when I refactored this code, all of this was basically happening in the Start method:
https://github.com/prysmaticlabs/prysm/blob/37ca409cc102f43360b65d816379a71ad73024c6/beacon-chain/blockchain/service.go

Comment (terencechain) [reply beacon-chain/blockchain/service.go]: I made a minor modification to remove `spawnProcessAttestationsRoutine` from within `startFromPOWChain` to use the `start` function instead. 

`startFromSavedState` is used in the newer implemented spec test to involve blockchain service, this is all after the your factor

Comment (kasey) [reply beacon-chain/blockchain/service.go]: one subtlety here - if you look at `spawnProcessAttestationsRoutine`, it actually subscribes to the StateFeed before spawning the goroutine, to make sure that this powchain goroutine and the attestation processing routine both subscribe before anything tries to read:
```
	stateChannel := make(chan *feed.Event, 1)
	stateSub := stateFeed.Subscribe(stateChannel)
```
inside of `startFromPOWChain`, right after `spawnProcessAttestationsRoutine`, you can see we start reading from stateChannel. If we spawn the goroutine in `startFromPOWChain`, and it starts reading the channel *before* we spawn the attestation processing routine, is there a possible race with the state channel? Could be fine, I would need to go read the StateFeed / notifier source to be confident in how all that works, just want to take note that we're changing order of operations on channel reads.

Comment (terencechain) [reply beacon-chain/blockchain/service.go]: I think it should be fine but please do double confirm once you get a chance, thanks!

**Review by kasey — APPROVED** [review]

**Pull request merged**